### PR TITLE
core: fix resource size calculation of cached images

### PR DESCRIPTION
**Pull request by g-yonchev**
**Summary**
This PR fixes an issue with cached images behaviour

**Related Issues/PRs**
see https://github.com/GoogleChrome/lighthouse/issues/12465#issuecomment-852896577

### CODE REVIEW

**Review by Janpot — APPROVED** [review]

**Comment by g-yonchev** [comment]
@googlebot I signed it!

File: lighthouse-core/lib/network-request.js [hunk 1]
```diff
@@ -495,6 +495,20 @@ class NetworkRequest {
         URL.isSecureScheme(record.protocol) ||
         URL.isLikeLocalhost(record.parsedURL.host);
   }
+
+  /**
+   * @param {NetworkRequest} networkRecord
+   * @return {number}
+   */
+  static getActualResourceSize(networkRecord) {
```
Comment (patrickhulce) [inline lighthouse-core/lib/network-request.js:503]: wdyt about `getTransferSizeOfResource` or `getResourceSizeOnNetwork` or something?

*actual resource size* implies to me that we're trying to get the raw size of the resource, but that's not what we're looking for. the point of the transfer size workaround is for cases where the image is compressed over the network again (svg especially). we're just trying to find the size of the image as it was transferred over the network without the headers if we can help it.

File: lighthouse-core/lib/network-request.js [hunk 2]
```diff
@@ -495,6 +495,20 @@ class NetworkRequest {
...
+    // Resource size is almost always the right one to be using because of the below:
```
Comment (patrickhulce) [inline lighthouse-core/lib/network-request.js:504]: move this into the jsdoc description for the function?

File: lighthouse-core/test/audits/byte-efficiency/uses-responsive-images-test.js [hunk 3]
```diff
@@ -226,4 +226,27 @@ describe('Page uses responsive images', () => {
...
+          generateSize(200, 200),
+          {width: 450, height: 450},
+          'https://google.com/logo.png',
+        ),
+      ],
+    },
+      [networkRecord]
+    );
+
+    assert.equal(auditResult.items.length, 1);
```
Comment (patrickhulce) [inline lighthouse-core/test/audits/byte-efficiency/uses-responsive-images-test.js:250]: assert that we're getting the wasted bytes correct too?

File: lighthouse-core/test/audits/byte-efficiency/offscreen-images-test.js [hunk 4]
```diff
@@ -554,4 +554,34 @@ describe('OffscreenImages audit', () => {
...
+          y: 0,
+          networkRecord,
+        }),
+      ],
+      traces: {defaultPass: createTestTrace({traceEnd: 2000})},
+      devtoolsLogs: {},
+    };
+
+    return UnusedImages.audit_(artifacts, [networkRecord], context).then(auditResult => {
+      assert.equal(auditResult.items.length, 1);
```
Comment (patrickhulce) [inline lighthouse-core/test/audits/byte-efficiency/offscreen-images-test.js:584]: assert that we're getting the wasted bytes correct too?

**Review by patrickhulce — COMMENTED** [review]
thanks very much @g-yonchev!

File: lighthouse-core/lib/network-request.js (hunk 1 above)
Comment (g-yonchev) [reply lighthouse-core/lib/network-request.js]: @patrickhulce, good point. I renamed the func to `getResourceSizeOnNetwork`.

**Review by patrickhulce — APPROVED** [review]
thanks very much @g-yonchev!

**Review by connorjclark — APPROVED** [review]

**Comment by Janpot** [comment]
@patrickhulce Just noticed that this PR didn't make it into 8.1.0. Are there any other criteria to be met for this PR to be merged?

**Comment by patrickhulce** [comment]
oh shoot, sorry @g-yonchev! No other criteria (other than fixing merge conflict now), just got forgotten in the shuffle

**Comment by g-yonchev** [comment]
I will fix the merge conflict.

**Comment by g-yonchev** [comment]
@patrickhulce I resolved the merge conflicts.
There was also a test failing which I fixed with the commit https://github.com/GoogleChrome/lighthouse/pull/12612/commits/b96e454d3b64b7c241f11db5a42d93a325e0a450

**Review by jamalxx1 — APPROVED** [review]

**Pull request merged**